src/site/content/en/payments/registering-a-web-based-payment-app/index.md

```diff
@@ -0,0 +1,321 @@
+
+## Configure a payment app with JavaScript
+
+Configuring a web-based payment app using JavaScript allows you to register
+multiple payment methods and manually unregister service workers.
+
+### Manually register a service worker
+
+A service worker is a JavaScript file you can register using
+[`navigator.serviceWorker.register()`](https://developer.mozilla.org/docs/Web/API/ServiceWorkerContainer/register).
+If you have a service worker file named `service-worker.js`, you can register it
+to the visitor's browser by running the following code:
+
+<!-- <div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe
+    src="https://glitch.com/embed/#!/embed/paymenthandler-demo?path=/public/index.js:32:1&attributionHidden=true&sidebarCollapsed=true"
+    alt="How the Glitch should look after completing step 1."
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div> -->
+
+{% Label %}[payment handler] payment app's landing page{% endLabel %}
+
+```js
+// Feature detection
+if ('serviceWorker' in navigator) {
+  // Register a service worker
+  navigator.serviceWorker.register(
+    // A service worker JS file is separate
+    'service-worker.js'
+  );
+  // PaymentManager requires the service worker to be active.
+  // One simple method to activate a service worker is through
+  // a `ready` promise.
+  const registration = await navigator.serviceWorker.ready;
+...
+```
+
+{% Aside %}
+Learn more about service worker registration in [Service Workers: an
+Introduction](https://developers.google.com/web/fundamentals/primers/service-workers).
+{% endAside %}
+
+### Set a payment instrument
+
+{% Aside 'key-term' %}
+Payment apps can support multiple ways for making a payment. For example, a
+customer can add multiple credit cards or e-wallets to a payment app. Each of
+them is a payment instrument.
+{% endAside %}
+
+Once the service worker is registered, [a service worker registration
+object](https://developer.mozilla.org/docs/Web/API/ServiceWorkerRegistration) is
+returned. Call
+[`registration.paymentManager.instrument.set()`](https://w3c.github.io/payment-handler/#paymentinstruments-interface)
+to set a payment instrument. It accepts two arguments, a string that represents
+the instrument key and a [`PaymentInstrument`
+object](https://w3c.github.io/payment-handler/#dom-paymentinstrument) that
+contains details about the instrument.
+
+{% Label %}[payment handler] payment app's landing pages{% endLabel %}
+
+```js
+...
+  // Feature detection
+  if (!registration.paymentManager) return;
+
+
+  await registration.paymentManager.instruments.set(
+    // Payment instrument key
+    "bobpay_card_1",
+    // Payment instrument details
+    {
+      // This parameter will be ignored in Chrome
+      name: 'Payment Handler Example',
+      // This parameter will be used to match against
+      // the PaymentRequest.
+      method: 'https://bobpay.xyz/pay'
+    }
+  );
+```
+
+<div class="w-table-wrapper">
```

Comment:
@robdodson this nested table looks a bit unorganized to me, can you think of any other way to represent it or make it look a bit more organized?

Reply:
@agektmr I will push up a commit proposing a way to make this table less complicated. I'm worried that it won't display correctly on mobile.